apps/examples/wifi_test/wifi_test.c

```diff
@@ -204,10 +204,10 @@ int wifi_test_proc(int argc, char *argv[])
 		memset(payload, 0, BUF_MAX_SIZE);
 
 		printScanResult(scan_info);
-		snprintf(payload, BUF_MAX_SIZE, "/%s|", DEVICE_NAME);
+		snprintf(payload, BUF_MAX_SIZE - 1, "/%s|", DEVICE_NAME);
 		for (i = 0; i < aps_cnt; i++) {
-			sprintf(buf, "%02d:%d/", g_aps[i].id, g_aps[i].rssi);
-			strncat(payload, buf, strlen(buf));
+			snprintf(buf, BUF_MAX_SIZE - 1, "%02d:%d/", g_aps[i].id, g_aps[i].rssi);
```

Comment:
It is little dangerous that reuse same buffer with out memset potentially.
Also snprintf cause Null termination issue, you can use strncpy.
But I think before copy buffer, you'd better check remained bytes and apply it if there is enough space.

Reply:
Since we use memset inside the loop, reuse should be safe. Also, as buf and payload have 0s by default using memset, snprintf does not have to care about appending '\0' at the end. Is this reasoning correct?

Reply:
I'm just suggesting you that something like below(as a example) to prevent non clear data insterting
if (offset + strlen(buf) > BUF_MAX_SIZE) {
    break;
}
If BUF_MAX_SIZE is calculated value based on size of id & rssi, you can ignore my comment.

Reply:
In this case, buf is always overwritten and there is no possiblity that the length of id plus rssi is greather than BUF_MAX_SIZE.
snprintf with the length of BUF_MAX_SIZE - 1 is enough without any security issues.

Reply:
Thanks for the explanation, valid point indeed. I think there needs to be a memset for buf inside the for loop. Also, id and rssi, along with the extra formatting should not cross 10 bytes, so buf need not be allocated 128 length. Something like char buf[BUF_MAX_SIZE>>3]; might work just fine. Memset inside the for loop is an overhead, but is also a simple fix. Just a suggestion from my side. Please go ahead with simpler alternatives if possible.

Reply:
If buf doesn't overflow, we don't need to clear buf because snprintf always inserts null termination.
The size of BUF_MAX_SIZE also doesn't matter because this is just an example.

Reply:
I add memset(buf) every iteration.